### ISSUE TITLE
ping(6) tests are failing on FreeBSD

### ISSUE THREAD

**Issue by wfurt** [dotnet/runtime#63530]
```
  System.Net.NetworkInformation.Tests.PingTest.SendPingWithIPAddress(addressFamily: InterNetworkV6) [FAIL]
        Assert.Equal() Failure
        Expected: Success
        Actual:   TimedOut
        Stack Trace:
          /usr/home/furt/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs(57,0): at System.Net.NetworkInformation.Tests.PingTest.PingResultValidator(PingReply pingReply, IPAddress[] localIpAddresses, ITestOutputHelper output)
          /usr/home/furt/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs(50,0): at System.Net.NetworkInformation.Tests.PingTest.PingResultValidator(PingReply pingReply, IPAddress localIpAddress)
          /usr/home/furt/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs(140,0): at System.Net.NetworkInformation.Tests.PingTest.<>c__DisplayClass7_0.<SendPingWithIPAddress>b__1(PingReply pingReply)
          /usr/home/furt/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs(791,0): at System.Net.NetworkInformation.Tests.PingTest.SendPing(Func`2 sendPing, Action`1 pingResultValidator)
          /usr/home/furt/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs(773,0): at System.Net.NetworkInformation.Tests.PingTest.SendBatchPing(Func`2 sendPing, Action`1 pingResultValidator)
          /usr/home/furt/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs(136,0): at System.Net.NetworkInformation.Tests.PingTest.SendPingWithIPAddress(AddressFamily addressFamily)
      System.Net.NetworkInformation.Tests.UnixPingUtilityTests.PacketSizeIsRespected(payloadSize: 0) [FAIL]
        System.Exception : [/sbin/ping6 -c 1 -x 1000 -s 16 ::1] process exit code is 1.
        StdOut:[]
        StdErr:[ping6: illegal option -- x
        usage:
        	ping [-4AaDdfHnoQqRrv] [-C pcp] [-c count] [-G sweepmaxsize]
        	    [-g sweepminsize] [-h sweepincrsize] [-i wait] [-l preload]
        	    [-M mask | time] [-m ttl][-P policy] [-p pattern] [-S src_addr]
        	    [-s packetsize] [-t timeout] [-W waittime] [-z tos] IPv4-host
        	ping [-4AaDdfHLnoQqRrv] [-C pcp] [-c count] [-I iface] [-i wait]
        	    [-l preload] [-M mask | time] [-m ttl] [-P policy] [-p pattern]
        	    [-S src_addr] [-s packetsize] [-T ttl] [-t timeout] [-W waittime]
                    [-z tos] IPv4-mcast-group
        	ping [-6aADdEfHnNoOquvyYZ] [-b bufsiz] [-c count] [-e gateway]
                    [-I interface] [-i wait] [-k addrtype] [-l preload] [-m hoplimit]
                    [-p pattern] [-S sourceaddr] [-s packetsize] [-t timeout]
        	    [-W waittime] [-z tclass] [IPv6-hops ...] IPv6-host
        ]
```

seems like regression from #35461. (cc: @sakno)

**Comment by Thefrank** [comment]
`/sbin/ping6 -c 1 -x 1000 -s 16 ::1` will work under FreeBSD 12.3 (or lower)
`/sbin/ping6 -c 1 -W 1000 -s 16 ::1` wo;; work under FreeBSD 13.0+

yes, PING changed between FreeBSD 12 and FreeBSD 13


`Merged the ping(8) and ping6(8) utilities. ping(8) supports both IPv4 and IPv6. A legacy ping6(8) is retained for backwards compatibility. 3cde9171d2d5`

legacy version does not take legacy options and args though which the error being encountered here [maybe?]

**Comment by sakno** [comment]
OMG, that looks like breaking change in `ping6` utility. Unfortunately, I was unable to predict that behavior. PR was merged more than year ago. @wfurt , we need to have both versions of FreeBSD in Azure DevOps pipeline for .NET Runtime.

**Comment by sakno** [comment]
Here is the code:
https://github.com/dotnet/runtime/blob/a9c5eadd951dcba73167f72cc624eb790573663a/src/libraries/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs#L91-L101

`OperatingSystem.IsFreeBSD()` is not enough anymore. Probably, `Environment.OSVersion` could help. However, on my Linux machine, [OperatingSystem.Version](https://docs.microsoft.com/en-us/dotnet/api/system.operatingsystem.version) property returning a kernel version, not a version of a distribution (Ubuntu in my case). `ping`/`ping6` is a user-mode programs and loosely coupled with kernel versions. I think the same situation on FreeBSD. So checking version via `Version` property can be potentially dangerous.

**Comment by wfurt** [comment]
yeah. It is really unfortunate that the syntax changed. and since we don't have regular tests it is really difficult to spot. 
I can check the  OperatingSystem.Version on. FreeBSD. Unlike Linux where you can have kernel and various distributions around it, FreeBSD is really unified OS.  We can also make the Version return what ever we feel is reasonable e.g. 12.x or 13.x in this case. 

Or we can just go with the 13.x flavor and document broken behavior on 12.

**Comment by sakno** [comment]
> Or we can just go with the 13.x flavor and document broken behavior on 12.

As you like, but I see no problem to maintain both versions. You can assign the issue to me. However, as I said, we need appropriate VMs in DevOps pipeline, because I'm working on Linux and have no chance to test it locally.

**Comment by Thefrank** [comment]
@sakno is getting VMs added something I can help out with or is that something that needs to be done internally?

**Comment by wfurt** [comment]
CI machines are not going to happen. (long story) I think ether me or @Thefrank are best sources for testing. 
I test my system and OperatingSystem.Version  returns 13.0.0 "norma" FreeBSD version.

**Comment by wfurt** [comment]
should be fixed with  #63531.
